File: 백준/Silver/1316. 그룹 단어 체커/그룹 단어 체커.py
n = int(input())  # 그룹단어의 개수를 입력한다.
# 비교 후 같지 않을 때만 조건문 걸어서 갯수를 하나 줄이도록 한다.

for _ in range(n):
    word = input()
    for i in range(len(word) - 1): # 앞 뒤 비교니까 -1
        if word[i] != word[i + 1]: # 앞 뒤 다를 경우 바로 비교
            if word[i] in word[i + 1:]: # 다른데 뒤에 해당 문자열 있으면 -1
                n -= 1
                break
print(n)

File: 백준/Silver/1316. 그룹 단어 체커/그룹 단어 체커.java
import java.util.ArrayList;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int count = sc.nextInt();
        int sum = 0;
        for (int i = 0; i < count; i++) {
            if (check(sc.next())) {
                sum++;
            }
        }
        System.out.println(sum);
    }

    public static boolean check(String str) {
        ArrayList<String> list = new ArrayList<>();
        for (int i = 0; i < str.length(); i++) {
            String word = str.substring(i, i + 1);
            if (!list.contains(word)) {
                list.add(word);
            } else if (str.charAt(i - 1) != str.charAt(i)) {
                // 리스트에 있는데 값이 다르면 false
                return false;
            }
        }
        return true;
    }
}
